lib/dataset/dataset.cpp

```diff
@@ -138,20 +140,19 @@ void Dataset::setCoord(const Dim dim, Variable coord) {
 void Dataset::setData(const std::string &name, Variable data,
                       const AttrPolicy attrPolicy) {
   expect_writable(*this);
-  setSizes(data.dims());
+  expect_valid(*this);
```

Comment:
Is it intentional that we check validity only in setters? Are we 100% certain that invalid datasets can never propagate to user code?

Reply:
No, we can always mess up. But where else do you want to check?

Reply:
So basically we might have an invalid Dataset that claims it has, e.g., empty sizes?

Reply:
Yes